src/raftstore/store/peer.rs

```diff
@@ -1893,10 +1935,28 @@ impl Peer {
         true
     }
 
-    pub fn get_min_progress(&self) -> u64 {
-        self.raft_group.status_ref().progress.map_or(0, |p| {
-            p.iter().map(|(_, pr)| pr.matched).min().unwrap_or_default()
-        })
+    pub fn get_min_progress(&self) -> Result<u64> {
+        let mut min = None;
+        if let Some(progress) = self.raft_group.status_ref().progress {
+            for (id, pr) in progress.iter() {
+                if pr.state == ProgressState::Snapshot
+                    || pr.pending_request_snapshot != INVALID_INDEX
+                {
+                    return Err(box_err!(
+                        "there is a pending snapshot peer {} [{:?}], skip merge",
```

Comment:
This error message is tricky, because get_min_progress is not only used for merge.